processing/src/main/java/io/druid/query/filter/DruidCompositePredicate.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.filter;
+
+import com.google.common.base.Predicate;
+
+/**
+ * Composite predicate class that can accept all supported types
+ *
+ * The apply() method inherited from Predicate<Object> is intended for String values
+ * and any other non-primitive Object types.
+ *
+ * The applyLong() method inherited from DruidLongPredicate is intended for long primitive values.
+ *
+ * A separate method is present for each supported primitive type to avoid boxing (for performance reasons)
+ */
+public interface DruidCompositePredicate extends Predicate<Object>, DruidLongPredicate
```

Comment:
@gianm ah, my bad, should've put a WIP on the last commit, I wasn't quite done with the PR changes yet but wasn't expecting you to review so soon :D